src/data_readers/data_reader_merge_features.cpp

```diff
@@ -43,7 +43,8 @@ data_reader_merge_features::data_reader_merge_features(
   const data_reader_merge_features& other) :
   generic_compound_data_reader(other),
   m_data_size(other.m_data_size) {
-  m_label_reader = other.m_label_reader->copy();
+  if(other.m_label_reader != nullptr) 
```

Comment:
We should handle the case where `other.m_label_reader == nullptr`.

Reply:
Fixed, thanks!